pkg/api/config.go

```diff
@@ -502,6 +505,22 @@ func validateCredentials(fieldRoot string, credentials []CredentialReference) []
 	return errs
 }
 
+func validateParameters(fieldRoot string, params []StepParameter, env TestEnvironment) error {
+	var missing []string
+	for _, param := range params {
+		if param.Default != "" {
+			continue
+		}
+		if _, ok := env[param.Name]; !ok {
+			missing = append(missing, param.Name)
+		}
+	}
+	if missing != nil {
+		return fmt.Errorf("%s: unresolved parameter(s): %s", fieldRoot, missing)
```

Comment:
This is user-facing, so I think this error should be much more explanatory and lead the user to the name of the step/workflow/chain that needs this parameter.

Reply:
https://github.com/openshift/ci-tools/pull/879 added contextual information to the errors (see `"unresolved test"` in the tests).

A more concrete example:

```
error: failed to load configuration: failed to resolve configuration: Failed resolve MultiStageTestConfiguration: e2e: ipi-aws-pre: ipi-install: unresolved parameter: FOO
```

Reply:
Oh, that's in `Validate`, not (the same error) in the resolver…

The only cases this covers is inline steps, in which case the problem should be pretty obvious.

Reply:
Yeah, I realized that after reading the rest of the PR (and of course only after submitting the review)...